partitioner/src/test/java/io/confluent/connect/storage/partitioner/TimeBasedPartitionerTest.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    @Test
+    public void testNestedRecordFieldTimeExtractor() throws Exception {
+        TimeBasedPartitioner<String> partitioner = new TimeBasedPartitioner<>();
+        Map<String, Object> config = createConfig("nested.timestamp");
+        partitioner.configure(config);
+
+        long timestamp = new DateTime(2015, 4, 2, 1, 0, 0, 0, DateTimeZone.forID(timeZoneString)).getMillis();
+        SinkRecord sinkRecord = createSinkRecordWithNestedTimeField(timestamp);
+
+        String encodedPartition = partitioner.encodePartition(sinkRecord);
+
+        assertEquals("year=2015/month=4/day=2/hour=1/", encodedPartition);
+    }
+
+    @Test
+    public void testRecordTimeExtractor() throws Exception {
+        TimeBasedPartitioner<String> partitioner = new TimeBasedPartitioner<>();
+        Map<String, Object> config = createConfig(null);
+        partitioner.configure(config);
+
+        long timestamp = new DateTime(2015, 4, 2, 1, 0, 0, 0, DateTimeZone.forID(timeZoneString)).getMillis();
+        SinkRecord sinkRecord = createSinkRecord(timestamp);
+
+        String encodedPartition = partitioner.encodePartition(sinkRecord);
+
+        assertEquals("year=2015/month=4/day=2/hour=1/", encodedPartition);
+    }
+
+    private static class BiHourlyPartitioner extends TimeBasedPartitioner<String> {
+        private static long partitionDurationMs = TimeUnit.HOURS.toMillis(2);
+
+        @Override
+        public String getPathFormat() {
+            return "'year'=YYYY/'month'=MMMM/'day'=dd/'hour'=H/";
+        }
+
+        @Override
+        public void configure(Map<String, Object> config) {
+            init(partitionDurationMs, getPathFormat(), Locale.ENGLISH, DATE_TIME_ZONE, config);
+        }
+    }
+
+    private static class HourlySchemaGenerator implements SchemaGenerator<String> {
+
+        private String delim;
+
+        public HourlySchemaGenerator(Map<String, Object> config) {
+            if (config == null) {
+                delim = StorageCommonConfig.DIRECTORY_DELIM_DEFAULT;
+            } else {
+                delim = (String) config.get(StorageCommonConfig.DIRECTORY_DELIM_CONFIG);
+                if (delim == null) {
+                    delim = StorageCommonConfig.DIRECTORY_DELIM_DEFAULT;
+                }
+            }
+        }
+
+        @Override
+        public List<String> newPartitionFields(String format) {
+            String[] partitionParts = format.split(delim);
+
+            List<String> fields = new ArrayList<>();
+
+            for (String field : partitionParts) {
+                String[] parts = field.split("=");
+                fields.add(parts[0].replace("'", ""));
+            }
+
+            return fields;
+        }
+    }
+
+
+    private Map<String, Object> createConfig(String timeFieldName) {
+        Map<String, Object> config = new HashMap<>();
+        config.put(PartitionerConfig.SCHEMA_GENERATOR_CLASS_CONFIG, HourlySchemaGenerator.class);
```

Comment:
I saw a commit for not making `SCHEMA_GENERATOR_CLASS_CONFIG` required, or it was removed? 

Not sure why that commit didn't get merged in here